daml-assistant/daml-helper/src/DamlHelper/Run.hs

```diff
@@ -84,53 +85,162 @@ data ReplaceExtension
     -- ^ Replace the extension if the current extension is newer.
     | ReplaceExtAlways
     -- ^ Always replace the extension.
+    | ReplaceExtPublished
+    -- ^ Replace with published extension (the default).
 
-runDamlStudio :: ReplaceExtension -> [String] -> IO ()
-runDamlStudio replaceExt remainingArguments = do
-    sdkPath <- getSdkPath
-    vscodeExtensionsDir <- fmap (</> ".vscode/extensions") getHomeDirectory
-    let vscodeExtensionName = "da-vscode-daml-extension"
-    let vscodeExtensionSrcDir = sdkPath </> "studio"
-    let vscodeExtensionTargetDir = vscodeExtensionsDir </> vscodeExtensionName
-    whenM (shouldReplaceExtension replaceExt vscodeExtensionTargetDir) $
-        removePathForcibly vscodeExtensionTargetDir
-    installExtension vscodeExtensionSrcDir vscodeExtensionTargetDir
-    -- Note that it is important that we use `shell` rather than `proc` here as
-    -- `proc` will look for `code.exe` in PATH which does not exist.
-    projectPathM <- getProjectPath
-    let codeCommand
-            | isMac = "open -a \"Visual Studio Code\""
-            | otherwise = "code"
-        path = fromMaybe "." projectPathM
-        command = unwords $ codeCommand : path : remainingArguments
-
+-- | Run VS Code command with arguments, returning the exit code, stdout & stderr.
+runVsCodeCommand :: [String] -> IO (ExitCode, String, String)
+runVsCodeCommand args = do
     -- Strip DAML environment variables before calling vscode.
     originalEnv <- getEnvironment
     let strippedEnv = filter ((`notElem` damlEnvVars) . fst) originalEnv
-        process = (shell command) { env = Just strippedEnv }
+        command = unwords ("code" : args)
+        commandEnv =
+            if isMac
+                then updateSearchPath strippedEnv -- can't assume code is in PATH on mac
+                else strippedEnv
+        updateSearchPath env =
```

Comment:
```suggestion
        addMacOSPath env =
```
Not a great name either but at least it makes it clear that it specific to MacOS.

Reply:
Thanks!

Reply:
I'd rather `addCodeToPath` which was a standalone function and had a if mac branch and a normal branch - then it has independent semantics.

Reply:
That made the code cleaner. Thanks!